src/utils.ts

```diff
@@ -21,3 +25,84 @@ export const remove0x = (str: string): string => {
     return str
   }
 }
+
+const wait = (milliseconds) => {
+  return new Promise((resolve) => setTimeout(resolve, milliseconds))
+}
+
+export async function sendTxWithITX(
+  itx: JsonRpcProvider,
+  signer: Signer,
+  to: string,
+  data: string,
+  gas: string,
+  iteration = 0
+): Promise<string> {
+  try {
+    const tx = {
+      to,
+      data,
+      gas: BigNumber.from(gas)
+        .add(50000)
+        .toString(), // A little extra gas since we fetched it using estimateGas
+    }
+
+    const relayTransactionHashToSign = keccak256(
+      defaultAbiCoder.encode(
+        ['address', 'bytes', 'uint', 'uint'],
+        [tx.to, tx.data, tx.gas, (await itx.getNetwork()).chainId]
+      )
+    )
+    const signature = await signer.signMessage(
+      arrayify(relayTransactionHashToSign)
+    )
+
+    return await itx.send('relay_sendTransaction', [tx, signature])
+  } catch (e) {
+    // Take into account JSON RPC errors that may arise (e.g. rate limiting)
+
+    if (iteration === RETRY_LIMIT) {
```

Comment:
Perhaps I'm paranoid, but I always use `>=` when comparing against a max, just in case someone does something funny with iteration.

Reply:
fixed